AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/CascadeDeleteOperation.swift

```diff
@@ -0,0 +1,489 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import Amplify
+import AWSPluginsCore
+import Combine
+
+enum DeleteInput {
+    case withId(id: Model.Identifier)
+    case withIdAndCondition(id: Model.Identifier, condition: QueryPredicate)
+    case withFilter(_ filter: QueryPredicate)
+
+    /// Returns a computed predicate based on the type of delete scenario it is.
+    var predicate: QueryPredicate {
+        switch self {
+        case .withId(let id):
+            return field("id").eq(id)
+        case .withIdAndCondition(let id, let condition):
+            return field("id").eq(id).and(condition)
+        case .withFilter(let filter):
+            return filter
+        }
+    }
+}
+
+public class CascadeDeleteOperation<M: Model>: AsynchronousOperation {
+
+    let storageAdapter: StorageEngineAdapter
+    var syncEngine: RemoteSyncEngineBehavior?
+    let modelType: M.Type
+    let modelSchema: ModelSchema
+    let deleteInput: DeleteInput
+    let completionForWithId: ((DataStoreResult<M?>) -> Void)?
+    let completionForWithFilter: ((DataStoreResult<[M]>) -> Void)?
+
+    private let serialQueueSyncDeletions: DispatchQueue
+
+    init(storageAdapter: StorageEngineAdapter,
+         syncEngine: RemoteSyncEngineBehavior?,
+         modelType: M.Type,
+         modelSchema: ModelSchema,
+         withId id: String,
+         condition: QueryPredicate? = nil,
+         completion: @escaping (DataStoreResult<M?>) -> Void) {
+        self.storageAdapter = storageAdapter
+        self.syncEngine = syncEngine
+        self.modelType = modelType
+        self.modelSchema = modelSchema
+        var deleteInput = DeleteInput.withId(id: id)
+        if let condition = condition {
+            deleteInput = .withIdAndCondition(id: id, condition: condition)
+        }
+        self.deleteInput = deleteInput
+        self.completionForWithId = completion
+        self.completionForWithFilter = nil
+        self.serialQueueSyncDeletions = DispatchQueue(label: "com.amazoncom.Storage.CascadeDeleteOperation.concurrency")
```

Comment:
This string can be a static member of the class

Reply:
what's the point of this? i'm hesitant to add this mainly because all other DispatchQueue's created across the codebase does not make a static string out of the label name, and is used in one place only.